src/operations/search_indexes/create.ts

```diff
@@ -0,0 +1,48 @@
+import type { Document } from 'bson';
+
+import type { Collection } from '../../collection';
+import type { Server } from '../../sdam/server';
+import type { ClientSession } from '../../sessions';
+import type { Callback } from '../../utils';
+import { AbstractOperation } from '../operation';
+
+/** @public */
+export interface SearchIndexDescription {
+  /** The name of the index. */
+  name?: string;
+
+  /** The index definition. */
+  description: Document;
+}
+
+/** @internal */
+export class CreateSearchIndexesOperation extends AbstractOperation<string[]> {
+  constructor(
+    private readonly collection: Collection<any>,
+    private readonly descriptions: ReadonlyArray<SearchIndexDescription>
+  ) {
+    super();
+  }
+
+  execute(server: Server, session: ClientSession | undefined, callback: Callback<any>): void {
```

Comment:
Do you mean typing it in `Callback`?

Reply:
Yep, for now at least that's where the result goes, when someone changes this to an async function I think it would be nice to have a "documented" result shape